packages/query-core/src/utils.ts

```diff
@@ -36,9 +35,7 @@ export interface QueryFilters<
   /**
    * Include queries matching this query key
    */
-  queryKey?: unknown extends TQueryFnData
-    ? QueryKey
-    : QueryKey & DataTag<unknown, TQueryFnData, TError>
```

Comment:
This was the wrong idea, any fully-typed usage will define the TQueryKey and pass it in, and this approach was proving unreliable when I added richer tests.